crates/bevy_math/src/interpolation/lerp.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{interpolation::utils::*, Quat, Vec2, Vec3, Vec4};
+
+/// Defines how a particular type will be interpolated
+pub trait Lerp: Sized {
+    /// Lerp, `t` is unclamped
+    fn lerp_unclamped(a: &Self, b: &Self, t: f32) -> Self;
+
+    /// Lerp, `t` is clamped in [0; 1] range
+    fn lerp(a: &Self, b: &Self, t: f32) -> Self {
+        let t = t.min(1.0).max(0.0);
```

Comment:
```suggestion
        let t = t.clamp(0.0, 1.0);
```
https://doc.rust-lang.org/std/primitive.f32.html#method.clamp